bin/check.ml

```diff
@@ -0,0 +1,70 @@
+open Bos_setup
+open Dune_release
+
+let assert_tag_exists repo tag =
+  if Vcs.tag_exists ~dry_run:false repo tag then Ok ()
+  else R.error_msgf "%s is not a valid tag" tag
+
+let clone_and_checkout_tag repo ~dir ~tag =
+  Sos.delete_dir ~dry_run:false ~force:true dir >>= fun () ->
+  Vcs.clone ~dry_run:false ~force:true repo ~dir >>= fun () ->
+  Vcs.get ~dir () >>= fun clone_vcs ->
+  Vcs.checkout ~dry_run:false clone_vcs ~branch:"dune-release-check"
+    ~commit_ish:tag
+
+let check (`Dist_name name) (`Package_names pkg_names)
+    (`Package_version version) (`Dist_tag tag) (`Skip_lint skip_lint)
+    (`Skip_build skip_build) (`Skip_tests skip_tests)
+    (`Working_tree on_working_tree) =
+  ( if on_working_tree then
+    OS.Dir.current () >>= fun dir ->
+    Check.check_project ~name ~pkg_names ~tag ~skip_lint ~skip_build ~skip_tests
+      ~dir
+  else
+    Vcs.get_tag tag version >>= fun inferred_tag ->
+    Vcs.get () >>= fun repo ->
+    assert_tag_exists repo inferred_tag >>= fun () ->
+    Fpath.of_string "_build/.dune-release-check" >>= fun dir ->
```

Comment:
Shouldn't _build be parameterized with `--build-dir` ?

Reply:
It totally should. Done. Thanks!